suite-common/wallet-transactions/src/transactionThunks.ts

```diff
@@ -0,0 +1,222 @@
+
+    const walletTransactions = selectTransactions(getState());
+
+    // find all transactions to replace, they may be related to another account
+    const transactions = findTransactions(tx.prevTxid, walletTransactions);
+    const newFee = formatNetworkAmount(tx.fee, account.symbol);
+    const newBaseFee = parseInt(tx.fee, 10);
+
+    // prepare replace actions for txs
+    const actions = transactions.map(t => {
+        const action = {
+            type: transactionActions.replaceTransaction.type,
+            payload: {
+                key: t.key,
+                txid: tx.prevTxid,
+                tx: {
+                    ...t.tx,
+                    txid: newTxid,
+                    fee: newFee,
+                    rbf: !!tx.rbf,
+                    blockTime: Math.round(new Date().getTime() / 1000),
+                    // TODO: details: {}, is it worth it?
+                },
+            },
+        };
+        // finalized and recv tx shouldn't have rbfParams
+        if (!tx.rbf || t.tx.type === 'recv') {
+            delete action.payload.tx.rbfParams;
+            return action;
+        }
+
+        if (action.payload.tx.type === 'self') {
+            action.payload.tx.amount = newFee;
+        }
+        // update tx rbfParams
+        if (action.payload.tx.rbfParams) {
+            action.payload.tx.rbfParams = {
+                ...action.payload.tx.rbfParams,
+                txid: newTxid,
+                baseFee: newBaseFee,
+                feeRate: tx.feePerByte,
+            };
+        }
+        return action;
+    });
+    // dispatch replace actions
+    actions.forEach(a => dispatch(a));
+});
+
+export const addFakePendingTxThunk = createAsyncThunk<
+    void,
+    {
+        precomposedTx: Pick<PrecomposedTransactionFinal | TxFinalCardano, 'totalSpent' | 'fee'>;
+        txid: string;
+        account: Account;
+    }
+>(`${modulePrefix}/addFakePendingTransaction`, ({ precomposedTx, txid, account }, { dispatch }) => {
+    // Used in cardano send form and staking tab until Blockfrost supports pending txs on its backend
+    // https://github.com/trezor/trezor-suite/issues/4932
+    const fakeTx = {
+        type: 'sent' as const,
+        txid,
+        blockTime: Math.floor(new Date().getTime() / 1000),
+        blockHash: undefined,
+        // amounts (as most of props below) don't matter much since it is temp fake anyway
+        amount: precomposedTx.totalSpent,
+        fee: precomposedTx.fee,
+        totalSpent: precomposedTx.totalSpent,
+        targets: [],
+        tokens: [],
+        cardanoSpecific: {
+            subtype: null,
+        },
+        details: {
+            vin: [],
+            vout: [],
+            size: 0,
+            totalInput: '0',
+            totalOutput: '0',
+        },
+    };
+    dispatch(transactionActions.addTransaction({ transactions: [fakeTx], account }));
+});
+
+export const prepareFetchTransactionsThunk = ({
```

Comment:
Use `createThunk` from redux-utils package everywhere in this file

Also you should be able to get rid of this "prepare" function because you have extra avaialable

Reply:
Fixed